pkg/reconciler/openshift/openshift.go

```diff
@@ -15,7 +18,7 @@ func reconcilerHook(cr *argoprojv1alpha1.ArgoCD, v interface{}) error {
 	switch o := v.(type) {
 	case *rbacv1.ClusterRole:
 		if o.ObjectMeta.Name == cr.ObjectMeta.Name+"-argocd-application-controller" {
-			if len(cr.Spec.ManagementScope.ClusterConfigNamespaces) > 0 {
```

Comment:
Let's keep the API knob, I think. Any reason you think we should not have it?

.spec.managementScope.cluster = true|false

Reply:
Also,

```
cr.ObjectMeta.Name+"-argocd-application-controller"
```

This doesn't give us very unique names, does it 🤔 . 

Maybe something like this could help us ?

```
cr.ObjectMeta.Name+"-"+cr.ObjectMeta.Namespace+"-argocd-application-controller"
```

Reply:
> ec.managementScope.cluster = true|false

I think this will provide an inconvenient advantage to the user. Users can request admins to add their namespace to the ARGOCD_CLUSER_CONFIG_NAMESPACES and may not bother about getting it removed if they dont need it in future. Also, I was thinking, why would a user request to add his namespace in ARGOCD_CLUSER_CONFIG_NAMESPACES and not add spec.managementScope.cluster = true in his CR.

Reply:
The admin would whitelist in the env var.
The user would then request for it in the CR. The request will be honored only if that namespace is whitelisted.

Reply:
We may choose to skip having the knob in the CRD as of now.

Reply:
@sbose78 Wouldn't this limit the length of the object name/namespace combination very severely, as you've already used 31 characters in the suffix and separator?